### PR TITLE
Add global coming soon property tracks

### LINKED ISSUE #47642: Add global coming soon mode property to wcadmin tracks events
## Issue Description

It would be useful to segment Tracks events by coming soon vs live mode.

## Acceptance Criteria

* Add a global `coming_soon` Tracks property to all wcadmin events.
* Use the values of:
  * `no` when the site / store is live.
  * `site` when the whole site is in coming soon mode.
  * `store` when only the store is in coming soon mode.

### PR DESCRIPTION
**Pull request by ilyasfoo**
### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/47642.

Adds `coming_soon` tracks property via `woocommerce_tracks_event_properties`

### How to test the changes in this Pull Request:

1. Set up a new store, make sure to **not skip** the core profiler so that tracking is enabled
1. Go to `WooCommerce > Settings > Site visibility`
4. Ensure `Coming soon` is selected and `Restrict to store pages only` is disabled
1. Go to `WooCommerce > Home`
2. Open up browser console and run `wcTracks.recordEvent('test_coming_soon', {})`
3. Go to networks tab and filter for `test_coming_soon`
4. Click on it and view the request payload
5. Observe `coming_soon: site` exists 
6. Go to `WooCommerce > Settings > Site visibility` and enable `Restrict to store pages only`
7. Refresh homescreen and run `wcTracks.recordEvent('test_coming_soon', {})`
5. Observe `coming_soon: store` exists 
6. Go to `WooCommerce > Settings > Site visibility` and set to `Live`
7. Refresh homescreen and run `wcTracks.recordEvent('test_coming_soon', {})`
5. Observe `coming_soon: no` exists 

image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Admin/Features/LaunchYourStore.php [hunk 1]
```diff
@@ -66,6 +67,30 @@ public function save_site_visibility_options() {
...
+	 *
+	 * @param array $event_properties Event properties array.
+	 *
+	 * @return array
+	 */
+	public function append_coming_soon_global_tracks( $event_properties ) {
+		if ( is_array( $event_properties ) ) {
+			$coming_soon = 'no';
+			if ( 'yes' === get_option( 'woocommerce_coming_soon', 'no' ) ) {
+				$coming_soon = 'yes';
```
Comment (psealock) [inline plugins/woocommerce/src/Admin/Features/LaunchYourStore.php]: I think you can remove this check. The options will always be "no"|"site"|"store" and never be "yes".

**Review by psealock — APPROVED** [review]
This is testing well and code looks good. I just made a small comment about unreachable result that can be simplified. Pre-approving

**Review by chihsuan — APPROVED** [review]
Except for Paul's comment, the code looks good to me. Thanks! 👍

File: plugins/woocommerce/src/Admin/Features/LaunchYourStore.php (hunk 1 above)
Comment (ilyasfoo) [reply plugins/woocommerce/src/Admin/Features/LaunchYourStore.php]: Great suggestion! Removed in [6259236](https://github.com/woocommerce/woocommerce/pull/47644/commits/62592366b3fb944081d38bf8500dc1f8b39725cb)

**Pull request merged**